### PR TITLE
Feat/add node truncate and truncate sync

### PR DESCRIPTION
**Pull request by majidsajadi**
Implemented `truncate` and `truncateSync` in `node/fs`. #648 

Ref to [node documents](https://nodejs.org/api/fs.html#fs_fs_truncate_path_len_callback):

> Passing a file descriptor is deprecated and may result in an error being thrown in the future

So I ignored handling file descriptors in implementation. let me know otherwise.

### CODE REVIEW

File: node/_fs/_fs_truncate_test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  },
+});
+
+Deno.test({
+  name: "SYNC: truncate entire file contents",
+  fn() {
+    const file: string = Deno.makeTempFileSync();
+    truncateSync(file);
+    const fileInfo: Deno.FileInfo = Deno.lstatSync(file);
+    assertEquals(fileInfo.size, 0);
```
Comment (kt3k) [inline node/_fs/_fs_truncate_test.ts:74]: @majidsajadi It seems that the errors from here will cause the leak of the temp file. How about surrounding these with `try { }` and calling `removeSync` in `finally {}`?

**Comment by kt3k** [comment]
@majidsajadi Thank you for your contribution! I left one comment.

**Comment by majidsajadi** [comment]
@kt3k thanks for your response. 
I fixed the issue I guess. and now I guess I should rebase with the main branch. let me know if the changes I have done are okay so I rebase the branch.

**Review by caspervonb — APPROVED** [review]
LGTM (failing tests are unrelated).

**Review by kt3k — APPROVED** [review]
@majidsajadi Thank you for updating 👍 LGTM.

The CI failure is not related to this branch as Casper says. Thank you for your contribution!

**Comment by kt3k** [comment]
blocked by #775

**Pull request merged**